cmd/edgectl/scout.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+type Scout struct {
+	mode       string
+	installID  string
+	newInstall bool
+}
+
+type ScoutMeta struct {
+	Key   string
+	Value interface{}
+}
+
+func NewScout(mode string) (*Scout, error) {
+	var installID string
+	var newInstall bool
+
+	// Get or create the persistent install ID for Edge Control
+	err := func() error {
+		// We store the persistent ID in ~/.config/edgectl/id to be consistent
+		// with Telepresence. We could instead use os.UserConfigDir() as the
+		// root of the destination, which may be the same on Linux, but is
+		// definitely different on MacOS.
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return errors.Wrap(err, "install ID")
+		}
+		configDir := filepath.Join(home, ".config", "edgectl")
+		if err := os.MkdirAll(configDir, 0755); err != nil {
+			return errors.Wrap(err, "install ID")
+		}
+		idFilename := filepath.Join(configDir, "id")
+
+		// Try to read an existing install ID
+		if installIDBytes, err := ioutil.ReadFile(idFilename); err == nil {
+			installID = string(installIDBytes)
+			// Validate UUID format
+			if _, err := uuid.Parse(installID); err == nil {
+				newInstall = false
+				return nil
+			} // else the value is not a UUID
+		} // else file doesn't exist, etc.
+
+		// Try to create a new install ID
+		installID = uuid.New().String()
```

Comment:
Instead of generating a UUID for identifying the install end-user, should we instead try to replicate what Ambassador does and marshall a UUID given the unique target installation namespace?

https://github.com/datawire/ambassador/blob/6a613b46ab10b060ef92b0d9546b22531188fa75/python/kubewatch.py#L178

This way, if a single edgectl binary is used to install multiple clusters, it'll result in multiple `installId`.

Reply:
The story specifically asks for 

1. Edge Control to have its own ID so that it can post something right away, before it establishes connectivity to a cluster
2. The installed Ambassador's cluster_id to be sent in an extra_trait

The second item is not implemented yet.

Reply:
Sending the Ambassador cluster_id in an extra_trait seems reasonable. As the code is working, let's ship it and review the collected events with @richarddli